config.py

```diff
@@ -511,5 +263,547 @@ def delete_password(self, account: str) -> None:
+
+        elif not isinstance(res, bool):
+            raise ValueError(f'__raw_get returned unexpected type {type(res)=} {res!r}')
+
+        return res
+
+    def set(self, key: str, val: Union[int, str, List[str], bool]) -> None:
+        """
+        Set sets the given key to the given value.
+
+        :param key: The key to set the value to
+        :param val: The value to set the key
+        """
+        self._settings[key] = val
+
+    def delete(self, key: str) -> None:
+        """
+        Delete the given key from the config.
+
+        :param key: the key to delete
+        """
+        del self._settings[key]
+
+    def save(self) -> None:
+        """Save the configuration."""
+        self._defaults.setPersistentDomain_forName_(self._settings, self.identifier)
+        self._defaults.synchronize()
+
+    def close(self) -> None:
+        """Close the configuration."""
+        self.save()
+        self._defaults = None
+
+
+class LinuxConfig(AbstractConfig):
+    """Linux implementation of AbstractConfig."""
+
+    SECTION = 'config'
+    # TODO: I dislike this, would rather use a sane config file format. But here we are.
+    __unescape_lut = {'\\': '\\', 'n': '\n', ';': ';', 'r': '\r', '#': '#'}
+    __escape_lut = {'\\': '\\', '\n': 'n', ';': ';', '\r': 'r'}
+
+    def __init__(self, filename: Optional[str] = None) -> None:
+        super().__init__()
+        # http://standards.freedesktop.org/basedir-spec/latest/ar01s03.html
+        xdg_data_home = pathlib.Path(os.getenv('XDG_DATA_HOME', default='~/.local/share')).expanduser()
+        self.app_dir = xdg_data_home / appname
+        self.app_dir.mkdir(exist_ok=True, parents=True)
+
+        self.plugin_dir = self.app_dir / 'plugins'
+        self.plugin_dir.mkdir(exist_ok=True)
+
+        self.respath = pathlib.Path(__file__).parent
+
+        self.internal_plugin_dir = self.respath / 'plugins'
+        self.default_journal_dir = None
+        self.identifier = f'uk.org.marginal.{appname.lower()}'  # TODO: Unused?
+
+        config_home = pathlib.Path(os.getenv('XDG_CONFIG_HOME', default='~/.config')).expanduser()
+
+        self.filename = config_home / appname / f'{appname}.ini'
+        if filename is not None:
+            self.filename = pathlib.Path(filename)
+
+        self.filename.parent.mkdir(exist_ok=True, parents=True)
+
+        self.config: Optional[ConfigParser] = ConfigParser(comment_prefixes=('#',), interpolation=None)
+
+        try:
+            self.config.read(self.filename)
+        except Exception as e:
+            logger.debug(f'Error occurred while reading in file. Assuming that we are creating a new one: {e}')
+            self.config.add_section(self.SECTION)
+
+        if (outdir := self.get_str('outdir')) is None or not pathlib.Path(outdir).is_dir():
+            self.set('outdir', str(self.home))
+
+    def __escape(self, s: str) -> str:
```

Comment:
No handy-dandy module for this sort of thing?

Are strings 'bare' in the file, or do we need to handle the relevant quote marks ?

Reply:
I was going to use regexp's escape, but there are some things it does that are weird on the outside, and yeah other than that they're bare